### ISSUE TITLE
WSL2 distros (docker-desktop, docker-desktop-data) won't stop when exiting Docker Desktop

### ISSUE THREAD

**Issue by ktzsolt** [docker/for-win#13938]
### Description

As title says, when I click exit docker desktop on the moby icon on the system tray or in the GUI bottom left green corner, then the GUI closes down as it should but the wsl2 docker distros are still running and they will not stop unless I do 
`wsl --shutdown` or 
`wsl --terminate docker-desktop` and `wsl --terminate docker-desktop-data`

```shell
C:\ wsl -l -v
  NAME                   STATE           VERSION
* ubuntu-22.04           Running         2
  docker-desktop-data    Running         2
  docker-desktop         Running         2
```

### Reproduce

1. Start Docker Desktop with WSL2 mode
2. Stop Docker Desktop
3. check the wsl distros with `C:\ wsl -l -v` are still running

### Expected behavior

WSL2 distros should be stopped after closing the Docker Desktop

### docker version

```bash
Client: Docker Engine - Community
 Version:           25.0.3
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        4debf41
 Built:             Tue Feb  6 21:13:09 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Desktop
 Engine:
  Version:          25.0.3
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       f417435
  Built:            Tue Feb  6 21:14:25 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    25.0.3
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     /usr/local/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.6-desktop.1
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.24
    Path:     /usr/local/lib/docker/cli-plugins/docker-debug
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.22
    Path:     /usr/local/lib/docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     /usr/local/lib/docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.1
    Path:     /usr/local/lib/docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.5.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-scout
WARNING: Plugin "/usr/local/lib/docker/cli-plugins/docker-scan" is not valid: failed to fetch metadata: fork/exec /usr/local/lib/docker/cli-plugins/docker-scan: no such file or directory

Server:
 Containers: 7
  Running: 3
  Paused: 0
  Stopped: 4
 Images: 55
 Server Version: 25.0.3
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 13.66GiB
 Name: docker-desktop
 ID: 2d16933a-1c6b-4b61-9269-f1eaca97abcf
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

7BC791B4-C209-493E-A4B3-7E60DB0ED42A/20240228120653

**Comment by michelolvera** [comment]
Same here, same version, it's not a big deal but it happens.

**Comment by MihaelaStoica** [comment]
Hey everyone, we fixed the issue in [Docker Desktop 4.29](https://docs.docker.com/desktop/release-notes/#4290), please give it a try. I'm closing this ticket, but if you encounter the same again, please let us know.

**Comment by ktzsolt** [comment]
Just tried it now, it looks good so far! thank you!

**Comment by rkohanyi** [comment]
For me it doesn't seem to work. I've just updated to 4.29.0.
I've recorded my experience, hope it helps, [available here as an `.mp4`](https://github.com/docker/for-win/assets/23233923/72bb2a34-a5eb-4fda-9409-c9235302eedb).

My environment.
- Windows 11 (10.0.22631 Build 22631)
- Docker Desktop 4.29.0 (145265)
- WSL 2.1.5.0

I've run a diagnostics from my terminal also, got this ID (if that helps): 371A162C-CB49-4088-942B-9EC396E76307/20240417134248
